### PR TITLE
Integrate with the API to get the API token

### PR DESCRIPTION
**Pull request by hieptl**
This PR fixes #

## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

## Description

Please do not review until this [PR](https://github.com/FairwindsOps/goldilocks/pull/501) is merged

### What's the goal of this PR?

* Integrate with the Insights API to get the API token.

### What changes did you make?

* Integrate with the below API endpoint

```js
POST @ /v0/oss/users

Request Body Example:
{
  "email": "oss.user@gmail.com",
  "project": "goldilocks"
}
```

### What alternative solution should we consider, if any?

* none

### CODE REVIEW

**Review by sudermanjr — CHANGES_REQUESTED** [review]
Minor change inline, holding off on further review until #501 is merged and this branch is rebased.

File: cmd/dashboard.go [hunk 1]
```diff
@@ -39,6 +40,7 @@ func init() {
 	dashboardCmd.PersistentFlags().BoolVar(&onByDefault, "on-by-default", false, "Display every namespace that isn't explicitly excluded.")
 	dashboardCmd.PersistentFlags().BoolVar(&showAllVPAs, "show-all", false, "Display every VPA, even if it isn't managed by Goldilocks")
 	dashboardCmd.PersistentFlags().StringVar(&basePath, "base-path", "/", "Path on which the dashboard is served.")
+	dashboardCmd.PersistentFlags().StringVar(&insightsHost, "insightshost", "https://insights.fairwinds.com", "Insights host the dashboard is connected.")
```
[the remark this answers is not in the document]
Comment (hieptl) [reply cmd/dashboard.go]: Hi Andy, I updated the code. Thank you for your feedback!

File: cmd/dashboard.go [hunk 2]
```diff
@@ -39,6 +40,7 @@ func init() {
 	dashboardCmd.PersistentFlags().BoolVar(&onByDefault, "on-by-default", false, "Display every namespace that isn't explicitly excluded.")
 	dashboardCmd.PersistentFlags().BoolVar(&showAllVPAs, "show-all", false, "Display every VPA, even if it isn't managed by Goldilocks")
 	dashboardCmd.PersistentFlags().StringVar(&basePath, "base-path", "/", "Path on which the dashboard is served.")
+	dashboardCmd.PersistentFlags().StringVar(&insightsHost, "insights-host", "https://insights.fairwinds.com", "Insights host the dashboard is connected.")
```
[the remark this answers is not in the document]
Comment (hieptl) [reply cmd/dashboard.go]: I updated the code. Thank you

File: pkg/dashboard/templates/dashboard.gohtml [hunk 3]
```diff
@@ -2,6 +2,12 @@
 <html lang="en-US">
 <head>
   {{ template "head" . }}
+  <script>
+    window.onload = function() {
+      window.INSIGHTS_HOST = "{{ .Data.InsightsHost }}"
+    }
```
Comment (rbren) [inline pkg/dashboard/templates/dashboard.gohtml:8]:
```suggestion
    window.INSIGHTS_HOST = "{{ .Data.InsightsHost }}"
```
we don't need the `onload`

**Comment by rbren** [comment]
I shouldn't be seeing the API key box yet, right? It just triggers the email?

**Review by rbren — APPROVED** [review]

**Comment by hieptl** [comment]
> I shouldn't be seeing the API key box yet, right? It just triggers the email?

Yes, the API Token section will be developed in another PR. Thank you!

**Pull request merged**